hedera-mirror-rest/transactions.js

```diff
@@ -786,18 +784,70 @@ const extractSqlFromTransactionsByIdOrHashRequest = async (transactionIdOrHash,
 
   if (nonce !== undefined) {
     params.push(nonce);
-    mainConditions.push(`${Transaction.NONCE} = $${params.length}`);
+    conditions.push(`${Transaction.NONCE} = $${params.length}`);
   }
 
   if (scheduled !== undefined) {
     params.push(scheduled);
-    mainConditions.push(`${Transaction.SCHEDULED} = $${params.length}`);
+    conditions.push(`${Transaction.SCHEDULED} = $${params.length}`);
+  }
+
+  return {conditions, scheduled, nonce};
+};
+
+const getTransactionsByTransactionIdsSql = (transactionKeys, filters, timestampField) => {
+  let payerAccountParams = [];
+  let params = [];
+
+  let minTimestamp = 0;
+  let idConditions = [];
+  let includePayer = false;
+
+  if (transactionKeys.length % 2 !== 0) {
+    throw new InvalidArgumentError('transaction keys must be in sequenced pairs of [payer,timestamp,payer,timestamp]');
+  }
+
+  for (let index = 0; index < transactionKeys.length; index += 2) {
+    let paramIndex = params.length;
+    const payer = transactionKeys[index];
+    const timestamp = BigInt(transactionKeys[index + 1]);
+
+    if (payer) {
+      payerAccountParams.push(`$${++paramIndex}`);
+      idConditions.push(`(${Transaction.PAYER_ACCOUNT_ID} = $${paramIndex} and 
+                               ${timestampField} = $${++paramIndex})`);
+      params.push(payer);
+      includePayer = transactionKeys;
+    } else {
+      idConditions.push(`${timestampField} = $${++paramIndex}`);
+    }
+    params.push(timestamp);
+
+    if (!minTimestamp || timestamp < minTimestamp) {
```

Comment:
initialize minTimestamp to `constants.MAX_LONG`, then simplify this to
```suggestion
    if (timestamp < minTimestamp) {
```